src/ace/managers/state.c

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <ace/managers/state.h>
+#include <ace/managers/log.h>
+
+/* Functions */
+
+#ifdef ACE_DEBUG
+void _checkPointers(const char *szPointerName, void *pPointer) {
```

Comment:
- you're not checking pointer**s** here
- name could be more meaningful, like `checkNull()`.
- fn could also take file/line args and be called via macro - see `blitCheck()`.
- I think you could also automatically generate pointer name as an arg: https://stackoverflow.com/questions/7561828/variable-name-from-macro-argument
- it could be declared as `static inline` and moved to ace/macros.h or something.

Reply:
Done, but this code can't be in `macros.h` in this shape - logs causing circular includes.